lib/spec/sww.sw-spec.js

```diff
@@ -0,0 +1,127 @@
+importScripts('/base/lib/spec/nodeMock.js');
+importScripts('/base/lib/sww.js');
+importScripts('/base/lib/router.js');
+
+describe('ServiceWorkerWare', function () {
+  'use strict';
+
+  var worker;
+  var request;
+
+  var fallbackMW = function(req, res) {
+    if (res !== null) {
+      return Promise.resovle(res);
+    }
+
+    return 'fallbackMW';
+  }
+
+  beforeEach(function() {
+    worker = new ServiceWorkerWare();
+    worker.init();
+    request = {
+      method: 'GET',
+      url: 'https://www.example.com',
+      clone: function() {}
+    };
+  });
+
+  describe('fallback mw', function(done) {
+
+    it('should fetch if no fallback middleware', function (done) {
+      var result = 'ok';
+      var stub = sinon.stub(self, 'fetch', function() {
```

Comment:
You can use `var stub = sinon.stub(self, 'fetch').returns(Promise.resolve(result));`